api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.util.Date;
+
+public class CohortMembership extends BaseOpenmrsData implements Comparable<CohortMembership> {
```

Comment:
The way you implemented the equals method doesn't follow what was decided on the design call, the semantics are different, membership A is equal to membership B only if they have the same uuid if you are basing equality on java semantics, what I'm trying to say is that membership A is equal to B if they match the same patient, start and end time. Therefore the contains method and calls like remove will misbehave because they are using equality based on java's semantics and that's not what we want